feat: resolve mapping authorizations Search authorization layer

maryarm opened the pull request:
## Description

In current state mapping's are not handled as Authorizations owner directly or indirectly. With this task, we should be able to apply authorizations in search queries when OIDC is enabled.

## Related issues

closes #30242

Commit:
feat: add zod library for form validation

hisImminence committed:
refactor: move javaHome code in own function

Commit:
feat: display error notification on failed login

Commit:
feat: add field and submit error validation for login page

Commit:
Revert "feat: add zod library for form validation"

This reverts commit a6a05f3688f01480568df3ccd750ef22dfe1ac1d.

Commit:
feat: Migrate ProcessInstance component to v2

Commit:
feat: migrate VariablesForm component to v2

lenaschoenburg committed:
feat: support group membership tracking

lenaschoenburg committed:
feat: support tenant membership tracking

lenaschoenburg committed:
refactor: store user tenant membership decoupled from tenant state

lenaschoenburg committed:
refactor: store user tenant membership decoupled from user state

lenaschoenburg committed:
test: remove invalidated tests

lenaschoenburg committed:
test: relax expected rejection message to not depend on tenant order

Commit:
feat: add playwright, setup initial configuration, add first test

nicpuppa committed:
fix: group key should be retrieved from the persisted object

This because in the command the key will not be present

nicpuppa committed:
refactor: remove assigned entity from group by groupId

nicpuppa committed:
refactor: get entities by groupId

nicpuppa committed:
test: adjust test to properly work with the ID on group remove entity

pedesen committed:
feat: migrate InputOutputMappings component to V2 API

pedesen committed:
test: mock processInstanceListenersStore to suppress test warnings

pedesen committed:
feat: set staleTime to Infinity

OmranAbazid committed:
refactor: use mappingId instead of key for mapping rules

closes #30445

Commit:
feat: add test cases to complete login test, add util, add path constants, general improvements

Commit:
feat: use path constants for routes

pedesen committed:
feat: migrate MetadataPopover component to v2 REST API

pedesen committed:
test: mock useProcessDefinitionXml in unrelated test

OmranAbazid committed:
feat: enable mapping-rules update

closes #29658

houssain-barouni committed:
refactor: remove Operate's schema manager

ChrisKujawa committed:
fix: improve archiver panels

 * Commit adds new archiver panels to see,
   count of queries and archiving batch size
 * Exporting again for external so some references to
   the datasource are fixed
 * Remove some wrongly added panels
 * Add a new panel in general, to better compare
   how many instances are completed and archived
 * Add a new panel to show archived process instances on avg

p-wunderlich committed:
feat: execute batch operation state to check if we can cancel it

p-wunderlich committed:
feat: extent batch operation state with cancel method

p-wunderlich committed:
test: add test for cancel batch operation in state

p-wunderlich committed:
feat: add processor for canceling batch operations

p-wunderlich committed:
feat: add new applier for batch operation cancel

p-wunderlich committed:
test: add processor tests for cancel batch operations

p-wunderlich committed:
feat: rdbms - for BO cancellation, also update status of remaining items if they are still active

p-wunderlich committed:
docs: add new batch operation item status "canceled"

p-wunderlich committed:
feat: extend rdbms exporter to cancel batch operations

p-wunderlich committed:
test: add test for cancel batch operations to rdbms exporter

ChrisKujawa committed:
feat: add new metric

 Add new metric for the archiving batch size,
 which is the count of completed process instances
 found via query

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda?pullRequest=30677) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda?pullRequest=30677) before we can accept your contribution.<br/>**10** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: jonathanlukas<br/>:white_check_mark: houssain-barouni<br/>:white_check_mark: OmranAbazid<br/>:white_check_mark: stoerti<br/>:white_check_mark: esraagamal6<br/>:white_check_mark: buccarel<br/>:white_check_mark: hisImminence<br/>:white_check_mark: panagiotisgts<br/>:white_check_mark: pedesen<br/>:white_check_mark: p-wunderlich<br/>:x: maryarm<br/>:x: Alex Bremard<br/>:x: abremard<br/><hr/>**Alex Bremard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda?pullRequest=30677) it.</sub>

pedesen committed:
test: replace store mocks by msw API mocks

pedesen committed:
test: replace query mocks by msw API mocks

p-wunderlich committed:
refactor: move execute relevant batch operation tests into one file

p-wunderlich committed:
feat: delete also chunks and pending entries for batch operations on delete

p-wunderlich committed:
feat: check in the batch operation scheduler if the batch still exists

ChrisKujawa committed:
refactor: use search query instead of aggregation

 * This commit replaces the previous used aggregation with
   a normal search query to filter for process instances or
   batch operations by end date
 * The use case for aggregations are completly different to what
   we wanted to use it for
 * Benchmarks has shown that aggregations perform way slower
   than a normal search query
 * It is not clear why we used an aggregration before
 * Aggregation are limited by how many results it can return,
   at somepoint it simply failed without any results, and
   unclear error messages. This made it hard to fine-tune.
 * The general change was motivated by
   819c0b70ac6356f1a49e0f7f3fb6ee33a8feee4c from @lenaschoenburg

Commit:
fix: match login page validation and error display to operate and tasklist, style fixes

Commit:
fix: move utils to avoid sharing code between e2e and app

Commit:
fix: remove unnecessary abstractions on tests

Commit:
fix: config improvements

Commit:
feat: prepare ProcessInstance component migration

Commit:
feat: prepare VariablesForm component migration

Commit:
feat: prepare ModificationDropdown component migration

pedesen committed:
test: replace hook and store mocks by msw API mocks

ChrisKujawa committed:
refactor: use search query instead of aggregation

 * Similar to
   0354a9693a092e81a50470b1ef0c2cff78bc6698 this commit
   affects the OS part of the archiving logic
 * This commit replaces the previous used aggregation with
   a normal search query to filter for process instances or
   batch operations by end date
 * The use case for aggregations are completly different to what
   we wanted to use it for
 * Benchmarks has shown that aggregations perform way slower
   than a normal search query
 * It is not clear why we used an aggregration before
 * Aggregation are limited by how many results it can return,
   at somepoint it simply failed without any results, and
   unclear error messages. This made it hard to fine-tune.
 * The general change was motivated by
   819c0b70ac6356f1a49e0f7f3fb6ee33a8feee4c from @lenaschoenburg

pedesen committed:
feat: migrate ExecutionToggleCount to v2 REST API

stoerti committed:
fix: add enum for batch operation type in REST API

stoerti committed:
feat: add process instance cancel batch operation command to CamundaClient #30661

ChrisKujawa committed:
refactor: remove unused rolloverIntervall

 * With removing the aggregations we no longer make use of the
   rolloverInterval
 * This commit removes the configuration, and all related references

cmur2 committed:
ci: add comment about GHA workflow owner

pedesen committed:
feat: migrate FlowNodeInstancesLog to v2 REST API

ChrisKujawa committed:
build: add missing dependency

 * New search query in ArchiverRepository
   makes use of JSON API to extract data/fields

stoerti committed:
feat: add batch operation API interfaces and types to camunda client #30715

stoerti committed:
feat: add batch operation GET implementation camunda client #30715

stoerti committed:
feat: add batch operation pi cancel E2E test #30715

pedesen committed:
test: fix visual regression tests

cmur2 committed:
ci: send detailed stats for RDBMS ITs

lenaschoenburg committed:
refactor: don't store group membership on user entities directly

lenaschoenburg committed:
test: re-enable test for user membership cleanup

p-wunderlich committed:
fix: Fixed wrap function of BatchOperationExecutionRecord

p-wunderlich committed:
feat: add new batch operation lifecycle record

p-wunderlich committed:
feat: add new batch operation lifecycle intent

lenaschoenburg committed:
refactor: store group tenant membership decoupled from group state

buccarel committed:
build: allow optimize loggers to be configured with env vars

p-wunderlich committed:
feat: register batch operation valueType lifecycle to exporters

nicpuppa committed:
fix: use groupId in the error message

stoerti committed:
feat: add unit test for batch operation query in camunda client #30715

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

maryarm committed:
feat: resolve mapping authorizations Search authorization layer

cmur2 committed:
ci: send detailed stats for RDBMS ITs (#30719)

## Description

As [discussed on
Slack](https://camunda.slack.com/archives/C071KP5BTHB/p1744092783596709)
we can extend coverage to this job.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)
-- workflow not on stable branches

## Related issues

None

panagiotisgts committed:
refactor: rename zeebeRecordsExportEnabled to exportLegacyRecords

cmur2 committed:
ci: define CODEOWNERS for many GHA workflows

maryarm committed:
fix: fix failing test

maryarm committed:
fix: fix failing test

maryarm committed:
fix: fix failing test

maryarm committed:
fix: fix failing test

maryarm committed:
fix: fix failing test

maryarm committed:
fix: fix failing test

panagiotisgts committed:
test: enable exportLegacyRecords in importer finished tests

cmur2 committed:
ci: add comment about reusable GHA action owner

pedesen committed:
test: replace query mocks by msw mocks

esraagamal6 committed:
refactor: add a template for requesting  test case for the core application test suite

nicpuppa committed:
Remove entity group processor/applier should work with groupId  (#30640)

## Description

<!-- Describe the goal and purpose of this PR. -->

Enable the processors/appliers to work with ID on group entity removal

## Related issues

closes #30029

houssain-barouni committed:
refactor: remove Operate's schema manager (#30682)

## Description

<!-- Describe the goal and purpose of this PR. -->
- Remove Operate's Schema manager in favor of the new unified schema
manager
- Temporarily moved `operate-user` index management to the unified
schema manager, until the consolidated-auth fully replaces the Operate
based authentication (`auth`)
- Operate `IndicesCheck` health check now uses the unified schema
manager to check if all the indices exist

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

relates to [#28899](https://github.com/camunda/camunda/issues/28899)

Commit:
fix: use carbon variables to match util styles

maryarm committed:
Merge remote-tracking branch 'origin/identity/30242-mapping-authz-search' into identity/30242-mapping-authz-search

Commit:
feat: add eslint playwright plugin

panagiotisgts committed:
test: fix RecordCounterTest

pedesen committed:
feat: migrate InputOutputMappings component to V2 API (#30648)

## Description

This PR is part of https://github.com/camunda/camunda/issues/30591 with
the goal to replace processInstanceDetailsDiagramStore by Tanstack
query.

#### InputOutputMappings component
- remove dependency to processInstanceDetailsDiagramStore
- add query to v2 XML API

#### Misc
- Adjust tests
- Set staleTime for process defnition xml to infinity, because the XML
for a given process definition key is static and will never change. This
avoids unnecessary re-fetches
- mock processInstanceListenersStore in test that are not related to
listeners to avoid console warnings

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

related to #30591

maryarm committed:
fix: remove useless dep

maryarm committed:
fix: remove useless dep

maryarm committed:
fix: remove useless dep

maryarm committed:
fix: remove useless dep

maryarm committed:
fix: remove useless dep

p-wunderlich committed:
test: add new command distribution idempotency test for batch operation lifecycle

houssain-barouni committed:
fix: syntax error in renovate.json

vitorwtavares committed:
Display error on login page to provide user feedback (#30402)

## Description

Display error on login page to provide user feedback

- Add notification on failed login
- Add validation for fields when one of them is empty on blur and on
submit

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #29849

houssain-barouni committed:
fix: syntax error in renovate.json (#30734)

## Description

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

esraagamal6 committed:
refactor: update labels

panagiotisgts committed:
docs: fix javadoc reference

pedesen committed:
feat: migrate ExecutionToggleCount and FlowNodeInstancesLog to v2 REST API (#30716)

## Description

This PR is part of https://github.com/camunda/camunda/issues/30591 with
the goal to replace processInstanceDetailsDiagramStore by Tanstack
query.

- migrate ExecutionCountToggle
- migrate FlowNodeInstanceLog
- adjust tests
- fix visual regression test, update snapshot (retry button in incident
table is now correctly shown)

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

related to #30591

cmur2 committed:
ci: document GHA workflow owners when known (#30729)

## Description

Based on my research I left in-file comments and added CODEOWNERS
entries for which workflows are in-scope of the Monorepo DevOps team.

The CODEOWNERS entries will lead to [PR review requests
automatically](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#about-code-owners)
which might be noisy -> we can reconsider that part.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

Related #30689

pedesen committed:
feat: migrate MetadataPopover component to v2 REST API (#30670)

## Description

This PR is part of https://github.com/camunda/camunda/issues/30591 with
the goal to replace processInstanceDetailsDiagramStore by Tanstack
query.

#### MetadataPopover/Details component
- remove dependency to processInstanceDetailsDiagramStore
- add query to v2 XML API

#### Misc

- Adjust tests
- Mock useProcessDefinitionXml in ListView test

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

related to #30591

maryarm committed:
Merge remote-tracking branch 'origin/identity/30242-mapping-authz-search' into identity/30242-mapping-authz-search

ChrisKujawa committed:
Improve archiving performance (#30684)

## Description

It was
[highlighted](https://camunda.slack.com/archives/C08D74J6HUG/p1740412731557659)
that the archiving is too slow to catch up with exporting/processing. We
were able to complete a lot of process instances, and also to export
them to ES. But they haven't been moved to the dated indices quickly
enough, causing to fill up ES at some point.

We can see in the following metric that we complete per partition ~50
process instances (on the medic benchmarks), but only ~5% are actually
moved to the dated indices, where they can be deleted by ILM later.

![archiver-lagging-behind](https://github.com/user-attachments/assets/354d052e-abc1-42b1-8526-4d43b74c0e58)

We haven't had not much observability around this topic and missed some
visualization (previously we added some panels via #30679). On top of
this, the PR adds some more panels to better visualize the archiving,
and how many instances are actually found to be archived.

This allowed better to pinpoint where the bottleneck was, querying the
completed process instances. Especially that it is not easy to fine-tune
related `rolloverBatchSize` as it would simply stop finding new results

![2025-04-07_14-46](https://github.com/user-attachments/assets/5106cf19-621f-4711-b7e4-946bd963ed48)

It turned out to be an issue with the used aggregation, also highlighted
by @lenaschoenburg
[previously](https://camunda.slack.com/archives/C08D74J6HUG/p1744033725997889?thread_ts=1744030148.764179&cid=C08D74J6HUG).

With the new metric and the
[solution](https://github.com/camunda/camunda/commit/819c0b70ac6356f1a49e0f7f3fb6ee33a8feee4c)
by @lenaschoenburg, using a search query instead of a aggregation, we
were able to recover the performance of the related benchmark

![ES-disk-going-down](https://github.com/user-attachments/assets/da2472b9-f7c2-43c9-acdd-05841e238f38)

![archive-complete-compared](https://github.com/user-attachments/assets/d077f135-cb53-4638-9fc9-68788ad4629b)

![archived-improved](https://github.com/user-attachments/assets/96b6f184-606e-4c36-ba69-eb63dbff43c4)

### Benchmark

When running the [benchmark of
night](https://grafana.dev.zeebe.io/d/chris-playground/chris-playground?orgId=1&var-DS_PROMETHEUS=prometheus&var-cluster=All&var-namespace=ck-archiving-count-benchmark&var-pod=All&var-partition=All&from=now-15m&to=now),
we were able to see that ES is able to delete more data again and the
archiver catched up see above.

![ES-disk-overtime](https://github.com/user-attachments/assets/65e7403b-563f-4f0d-b178-250a136f9b9b)

![archiving-overtime](https://github.com/user-attachments/assets/3b5f4c6b-e77b-4ab5-9ec8-ddb667a70bb5)

Super interesting to see the PI completion and archiving compared

![pi-completion-vs-archving](https://github.com/user-attachments/assets/da2a9726-d13b-408f-97e9-4ac0f895a01e)

maryarm committed:
Merge remote-tracking branch 'origin/identity/30242-mapping-authz-search' into identity/30242-mapping-authz-search

buccarel committed:
fix: update -> upgrade

stoerti committed:
Feat/30661 start pi cancel by camunda client (#30712)

## Description

This adds the first batchOperation creation command to the Camunda
Client. The API currently uses two steps:

```
camundaClient
  .newCreateBatchOperationCommand()
  .processInstanceCancel()
  .filter(filter)
  .send();
```

1. The first step creates a generic CreateBatchOperationCommand
2. The second step then specializes the command to a specific type and
can apply a filter
3. possible additional generic steps can be added after this until the
command is sent to the REST gateway

Note, that currently only the REST gateway is supported. Currently we
have no GRPS representation of any search filter available.

The REST API also introduces a new BatchOperationTypeEnum (instead if
the previous String). This is used to temporarily store the type in the
command builder.

## Related issues

closes #30661

Commit:
fix: remove playwright eslint plugin

vitorwtavares committed:
Set up playwright for E2E tests and create first test (#30468)

## Description

- Adds playwright to Identity FE and configure E2E tests.
- Adds first test to validate test suite is working.

### TODO
- [x] Install Playwright library in frontend
- [x] Initial configuration for playwright
- [x] Initial setup for `playwright.config.ts` file to have test suite
work as intended for our purposes
  - [x] Adjusting ESLint rules for E2E context
- [x] Create first test - Login page test
  - [x] Create basic version with just one test case
- [x] Complement test suite to match all relevant test cases included in
Operate and Tasklist's login page tests
- [x] Run tests for validation
  - [x] Run basic login test locally successfully
  - [x] Run complete test locally successfully

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #30465

Commit:
refactor: use constant and improve tests

jonathanlukas committed:
test: assert that camunda client is disabled

jonathanlukas committed:
test: add legacy compatibility tests

stoerti committed:
feat: add enum check for batch operation enums in camunda-client #30715

p-wunderlich committed:
refactor: fixes name of BatchOperationLifecycleManagementRecord

jonathanlukas committed:
test: assert that camunda client is disabled (#30776)

## Description

<!-- Describe the goal and purpose of this PR. -->

This PR adds a test to assert that the `camunda.client.enabled` flag
actually works.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

stoerti committed:
fix: rename batch operation status to state in search API and RDBMS #30781

houssain-barouni committed:
test: make sure camunda app is stopped on test failre

OmranAbazid committed:
refactor: use mappingId instead of key for mapping rules (#30649)

## Description
<!-- Describe the goal and purpose of this PR. -->
See #30445

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #30445

maryarm committed:
Merge remote-tracking branch 'origin/identity/30242-mapping-authz-search' into identity/30242-mapping-authz-search

OmranAbazid committed:
feat: enable mapping-rules update  (#30678)

Please review this first: https://github.com/camunda/camunda/pull/30649

## Description

<!-- Describe the goal and purpose of this PR. -->
see parent issue #29658

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #29658

buccarel committed:
build: remove zeebe-build-tools from Optimize

koevskinikola committed:
refactor: normalize user group membership (#30529)

Similar to https://github.com/camunda/camunda/pull/30451, this removes
group membership from the user entity and group state and uses the
generalized membership state instead.

closes #30453

esraagamal6 committed:
Create a template for requesting a test case for the core application e2e test suite (#30732)

## Description

This PR introduces a new **E2E Test Case Template** for the **Core
Application** automated test suite. The template is designed to document
test cases for the E2E tests with clear sections on the **test flow**,
**preconditions**, **test data**, and **postconditions**. The template
aims to standardize test case documentation for more efficient and
consistent test creation.

### Key updates:
- Added a new template for E2E test cases
- The template includes sections for:
  - Description
  - Preconditions
  - Test Data (mandatory)
  - User Story (in Gherkin format)
  - Postconditions
  - Supported Versions
  - Priority
  - Additional Information
  - Definition of Ready checklist

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

houssain-barouni committed:
test: make sure camunda app is stopped on test failure (#30792)

## Description

<!-- Describe the goal and purpose of this PR. -->
As discussed in
[Slack](https://camunda.slack.com/archives/C071KP5BTHB/p1744125772038569)
`PrefixMigrationIT.shouldMigrateCorrectIndicesDuringPrefixMigration` is
not idempotent for retries on test failures.
The problem may come from the app not being stopped when there are
assert failures.
I added a finally block to make sure to stop the app on assertion
failures

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

relates to https://github.com/camunda/camunda/issues/30782

koevskinikola committed:
refactor: normalize user tenant membership (#30594)

Builds on top of https://github.com/camunda/camunda/pull/30529 and
implements the same change for user tenant membership.

:warning: I've also removed some checks that the assigned user actually
exists. This will no longer be required.

closes #30454

stoerti committed:
fix: rename batch operation status to state in REST API and camunda-client #30781

koevskinikola reviewed (changes requested):
❌ Hey @maryarm, are the changes in this PR correct? I see 78 commits and 242 changed files. A lot of them don't seem related to the GitHub issue description.

Can you rebase this branch and ensure that only the relevant changes are included?

stoerti committed:
Feat/30715 query batch operations camunda client (#30726)

## Description

This PR introduces batch operation GET requests to the camunda client:
- GET request to read a single batch operation
- GET request to read the items of a single batch operation

Also a first round-trip E2E for batch operations test using the
camunda-client is added.

## Related issues

closes #30715

p-wunderlich committed:
feat: extend state to pause a batch operation

p-wunderlich committed:
feat: add new processor to pause a batch operation

p-wunderlich committed:
feat: add new applier for paused batch operations

p-wunderlich committed:
feat: check if a batch operation is paused in relevant processors / tasks

p-wunderlich committed:
test: provide a processor test for pausing batch operations

p-wunderlich committed:
feat: export the paused batch operation status in rdbms

p-wunderlich committed:
test: integration test for paused batch operation in rdbms

koevskinikola committed:
refactor: store group tenant membership decoupled from group state (#30725)

Builds on top of https://github.com/camunda/camunda/pull/30594 and
implements the same change for group tenant membership.

closes #30458

Commit:
refactor: rename token operations const

abremard committed:
feat: prepare ProcessInstance component migration (#30702)

## Description

This PR is copying the v1 files to v2 subfolder to prepare the migration
in #30413

## Related issues

related to #30264

p-wunderlich committed:
feat: add CANCELED to BatchOperationItemState

p-wunderlich committed:
feat: cancel batch operation in engine (#30685)

## Description

* Adds new processor and applier in order to process the newly
introduced batch operation cancel intent
* Extends RDBMS Exporter and DB to store if a batch operation get's
canceled
* Since it could be, that there are still items which got not processed
until the cancel occurs, we now also set a CANCELED state to batch
operation items.

## Related issues

closes #29888

esraagamal6 committed:
fix: Update test-case.md

esraagamal6 committed:
refactor: rename test-case.md to core_application_test_request.md

esraagamal6 committed:
refactor: rename test-case.md to core_application_test_request.md

p-wunderlich committed:
test: add new command distribution idempotency test for batch operation pause lifecycle

esraagamal6 committed:
fix: Update test-case.md (#30818)

## Description
Running `mvn spotless:apply` caused formatting issues in the template.
This PR fixes those issues and also renamed the file to reflect what the
template does

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

panagiotisgts committed:
pg-legacy-record-fixes (#30728)

## Description
- Rename `zeebeRecordsExportEnabled` to `exportLegacyRecords` to be more
descriptive
- Enable the `exportLegacyRecords` in Importer Finished test cases
<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

p-wunderlich committed:
fix: Don't append execution command after fail

pedesen committed:
feat: migrate ProcessInstanceHeader to v2 XML API

p-wunderlich committed:
refactor: Provide better rejection handling for pausing a batch operation

p-wunderlich committed:
test: Add additional state tests for batch operation pause

buccarel committed:
build: remove zeebe-build-tools from Optimize (#30799)

## Description

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

stoerti committed:
Fix/30781 align batch operation status enums (#30801)

## Description

This PR fixes naming inconsistencies of the batch operation status/state
fields in search-api, RDBMS impl, Rest API and camunda-client

## Related issues

closes #30781

abremard committed:
feat: Migrate ProcessInstance component to v2 (#30413)

## Description

In this PR, I'm migrating `ProcessInstance` component away from
`processInstanceDetailsStatisticsStore` :
- I've removed statistics fetching and polling (all the child components
that rely on the polling will be migrated in later PRs)
- Updated the tests
- Added the v2 feature flag in the parent component

## Related issues

closes #30264

abremard committed:
feat: prepare VariablesForm component migration (#30703)

## Description

This PR is copying the v1 files to v2 subfolder to prepare the migration
in #30419

## Related issues

related #30265

esraagamal6 committed:
fix: Update core_application_test_request.md

abremard committed:
feat: migrate VariablesForm component to v2 (#30419)

## Description

In this PR, I'm migrating `VariablesForm` away from
`processInstanceDetailsStatisticsStore` :
- introduced new hooks to replace old flownode instance statistics logic
- updated tests
- updated top level component with feature flag

When testing the component locally I'm seeing a loading icon whenever I
change or add a variable. I've tested in v1 and the behavior is the
same, just wanted to point that out as my Operate knowledge is still
limited

https://github.com/user-attachments/assets/3f15666f-5884-44d3-85a4-8918a77d204f

## Related issues

closes #30265

buccarel committed:
fix: logging init

pedesen committed:
feat: migrate ProcessInstanceHeader to v2 XML API (#30827)

## Description

- migrate ProcessInstanceHeader to v2 XML API

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

related to #30591

esraagamal6 committed:
refactor: update labels

esraagamal6 committed:
fix: add correct labels in the core application test e2e request issue template (#30824)

## Description
Within the previous PR, `kind/bug` label was added by mistake, this PR
adds the correct label `kind/test-case`
<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

p-wunderlich committed:
feat: Also validate the distributed pause batch operation command

buccarel committed:
fix: update readme

p-wunderlich committed:
test: fix testing of failed batch operation pause command in process tests

buccarel committed:
fix: update readme

buccarel committed:
build: allow optimize loggers to be configured with env vars (#30724)

## Description

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

p-wunderlich committed:
test: Add additional test if pause batch operation works correct in the executor

p-wunderlich committed:
feat: pause batch operation in engine (#30819)

## Description

* Adds new processor and applier in order to process the newly
introduced batch operation pause intent
* Extends RDBMS Exporter and DB to store if a batch operation get's
paused

## Related issues

closes #29889

hisImminence committed:
refactor: move javaHome code in own function (#30700)

/closes
https://github.com/orgs/camunda/projects/33/views/1?pane=issue&itemId=102576667&issue=camunda%7Ccamunda%7C29788

## Description

<!-- Describe the goal and purpose of this PR. -->

moving java home related code into its own function.

At this point its too easy to be a good first issue anymore for Eamonn
:P

## Checklist

<!--- Please delete options that are not relevant. Boxes should be
checked by reviewer. -->
- [ ] for CI changes:
- [ ] structural/foundational changes signed off by [CI
DRI](https://github.com/cmur2)
- [ ]
[ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml)
modifications comply with ["Unified CI"
requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
- [ ] enable backports [when
recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

abremard committed:
feat: prepare ModificationDropdown component migration (#30704)

## Description

This PR is copying the v1 files to v2 subfolder to prepare the migration
in #30491

## Related issues

related #30268

maryarm committed:
Merge remote-tracking branch 'origin/identity/30242-mapping-authz-search' into identity/30242-mapping-authz-search

maryarm commented:
Hey @koevskinikola, you're right. I don't know hat happened during rebase but I'm going to close this PR and create a new branch to see if the problem will be solved.
